core/src/autogluon/core/metrics/__init__.py

```diff
@@ -104,17 +104,20 @@ def __call__(self, y_true, y_pred, sample_weight=None):
             y_true = np.array(y_true)
         if isinstance(y_pred, list):
             y_pred = np.array(y_pred)
-        type_true = type_of_target(y_true)
 
-        if len(y_pred.shape) == 1 or y_pred.shape[1] == 1 or type_true == 'continuous':
+        if len(y_pred.shape) == 1 or y_pred.shape[1] == 1:
             pass  # must be regression, all other task types would return at least two probabilities
-        elif type_true in ['binary', 'multiclass']:
-            y_pred = np.argmax(y_pred, axis=1)
-        elif type_true == 'multilabel-indicator':
-            y_pred[y_pred > 0.5] = 1.0
-            y_pred[y_pred <= 0.5] = 0.0
         else:
-            raise ValueError(type_true)
+            type_true = type_of_target(y_true)
```

Comment:
This change avoids computing type_true until necessary, speeds up regression metric calculation by ~50% in a lot of cases.